pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -545,17 +549,19 @@ func getJmsConfig(environment v1.EnvironmentType, jms *v1.KieAppJmsObject) (*v1.
 		KieServerJmsQueueSignal:        "queue/KIE.SERVER.SIGNAL",
 		KieServerJmsEnableAudit:        false,
 		KieServerJmsQueueAudit:         "queue/KIE.SERVER.AUDIT",
-		KieServerJmsAuditTransacted:    true,
+		KieServerJmsAuditTransacted:    t,
 		KieServerJmsUsername:           "user" + string(shared.GeneratePassword(4)),
 		KieServerJmsPassword:           string(shared.GeneratePassword(8)),
 	}
 
+	mergo.Merge(defaultJms, jms, mergo.WithOverride)
```

Comment:
@spolti i wonder if the addition of this line is what's causing the template problem?

mergo is already executed a few lines later...
https://github.com/kiegroup/kie-cloud-operator/blob/4a610088e81903d79fb1151a52a7ff910878df46/pkg/controller/kieapp/defaults/defaults.go#L576

Reply:
Yes, I did this to firstly, merge the jms values on to defaultJms values overriding all values set.
the second is used only the apply the Queues parameter and, if I return the defaultJms instead the jms struct, the operator didn't work, the AMQ pod never gets up. not sure why.

Reply:
@spolti to figure this out, would you please create a new branch, and in there, create a unit test that reproduces the original problem with the true/false default? Let us look at that to understand the problem, and if this is the best/right solution.

Reply:
@bmozaffa a small reproducer can be found with the changes on this branch: https://github.com/spolti/kie-cloud-operator/tree/temp